02-express-tutorial/app.js

```diff
@@ -1,10 +1,99 @@
 const express = require("express");
-const { products } = require("./data");
+const { products,people } = require("./data");
+const  peopleRouter  = require("./routes/people");
+const cookieParser = require("cookie-parser");
+
 const app = express();
-app.use(express.static("./public"));
+app.use(cookieParser());
+// Checking if req.cookies.name is present
+const auth = (req, res, next) => {
+if (req.cookies.name){
+  console.log(req.cookie.name);
+  req.user = req.cookies.name;
+  next();  // calling next method
+} else {
+  res.status(401).json({ message : 'unauthorized' });
+}
+};
+
+
+//  req => middleware => res
+const logger = (req, res, next) => {
+  const method = req.method
+  const url = req.url
+  const time = new Date().getFullYear()
+  console.log(method, url, time)
+  // res.send("testing")
+  next()
+}
+// app.use(logger);
+// static assets
+app.use(express.static('./methods-public'))
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+app.use("/api/v1/people", peopleRouter);
```

Comment:
There's a mismatch between this and the javascript.html, it uses POST /api/people instead of /api/v1/people, so it can't find the request. Make sure to test your code!

Reply:
Thank you, I will make sure to test my code.